cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataHttpHandler.java

```diff
@@ -526,7 +526,11 @@ public void removeStreamTags(HttpRequest request, HttpResponder responder,
       throw new BadRequestException("Unable to read business metadata from request.");
     }
     try (Reader reader = new InputStreamReader(new ChannelBufferInputStream(content), Charsets.UTF_8)) {
-      return GSON.fromJson(reader, MAP_STRING_STRING_TYPE);
+      Map<String, String> metadata = GSON.fromJson(reader, MAP_STRING_STRING_TYPE);
```

Comment:
not related to your change, but i think it makes sense to try ``fromJson`` for invalid JSON, instead of throwing the exception, currently we return Internal_Server_Error on an invalid json format, while we should return Bad_Request.